applications/script.js

```diff
@@ -73,9 +81,16 @@ function closeApplicationDetails() {
   applicationDetailsModal.classList.add('hidden');
   backDropBlur.style.display = 'none';
   document.body.style.overflow = 'auto';
+  window.history.replaceState(window.history.state, '', '/applications');
 }
 
-function openApplicationDetails(application) {
+function openApplicationDetails(application, renderById) {
+  if (!renderById) {
+    const currentUrlParams = new URLSearchParams(window.location.search);
+    currentUrlParams.append('id', application.id);
+    const applicationByIdUrl = '/applications/?' + currentUrlParams.toString();
+    window.history.replaceState(window.history.state, '', applicationByIdUrl);
+  }
   currentApplicationId = application.id;
```

Comment:
`openApplicationDetails(application, renderById)` function is overloaded with responsibilities—managing both the application data rendering and URL state. 



Split the function into smaller functions, one for managing the URL state and another for rendering the application details.

Reply:
Good point. Extracted 2 methods to add and remove query param from the url and update it